pkg/storage/stores/shipper/compactor/compactor.go

```diff
@@ -408,7 +409,7 @@ func (c *Compactor) CompactTable(ctx context.Context, tableName string, applyRet
 	interval := retention.ExtractIntervalFromTableName(tableName)
 	intervalMayHaveExpiredChunks := false
 	if c.cfg.RetentionEnabled && applyRetention {
-		intervalMayHaveExpiredChunks = c.expirationChecker.IntervalMayHaveExpiredChunks(interval)
+		intervalMayHaveExpiredChunks = c.expirationChecker.IntervalMayHaveExpiredChunks(interval, "")
```

Comment:
Should this still be decided here ? it seems that we don't have the user id for that table yet.

May be this is for supporting all format.

Reply:
When we pass empty user id to `IntervalMayHaveExpiredChunks`, it returns back true if either of the default retention period or any of the user's retention period causes data to be deleted within the given interval. We again check it for specific users index to see whether they may have expired chunks https://github.com/grafana/loki/blob/6bb44bceee8d2db9637da93ffd1f13f240c90cb4/pkg/storage/stores/shipper/compactor/table.go#L207